spec/threema/receive/delivery_receipt_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Threema::Receive::DeliveryReceipt do
+  subject { described_class.new(**attributes) }
+  let(:attributes) { attributes_for(:delivery_receipt_receive) }
+
+  context '#content' do
+    it { is_expected.to respond_to(:content) }
+
+    it 'returns the content' do
+      expect(subject.content).to eq(attributes[:content])
+    end
+  end
+
+  context 'message ids' do
+    let(:attributes) { { content: "\x01\xE0\xC8\x12n8]\xF3\x19J\xB4+ \xC9\xAB|\xB6\xAAc#\v|Ro(".b } }
+
+    it 'returns an array of message ids' do
+      expect(subject.message_ids).to be_a(Array)
+    end
+
+    it 'with each message id' do
+      expect(subject.message_ids.length).to eq(3)
+    end
+
+    it 'with correct length' do
+      expect(subject.message_ids.all? { |string| string.length.eql?(16) }).to be(true)
+    end
+  end
+
+  context 'timestamp' do
+    it 'returns a timestamp' do
+      expect(subject.timestamp).to be_an_instance_of(Integer)
+      expect(Time.at(subject.timestamp, in: 'UTC')).to be_an_instance_of(Time)
```

Comment:
It would be nice to have an explicit test case here, where you have the timestamp as string and then parse it.

One could consider to use `timecop` to make this test independent of the system time.

However, I wouldn't add the dependency for just one test case though.

Reply:
I also have questions about this. I feel like what I want to test here is not that the time is `Time.now`, which I could achieve with `Timecop.freeze`. I simply want to test that the consumer will receive an integer, that represents the time that they can either store as is, or convert to a dateime object.

Is there some reason, I should favor having it be exact? One advantage could be that if someone refactors the code and changes it to set the timestamp to `5.days.ago.to_i` the current test would still pass, but I don't think that is likely.

Reply:
I just like to have literals as assertion values. It's quicker to read. Take `[1,2,3]` as an example. You see immediately that it is of type array, is 3 items long and all numbers. That's it.

By the way "the consumer will receive an integer, that represents the time that they can either store as is, or convert to a dateime object." could be a great test description.